### PR TITLE
15 add cart and wishlist functionality

### LINKED ISSUE #15: Add cart and wishlist functionality.
- Change useAxios hook for cart and wishlist request.
- Add product display page
- Cart management for logged in user
- Wishlist management for logged in user

### PR DESCRIPTION
**Pull request by rdshinde**
- Changed useAxios for cart and wishlist functionality.
- Added AddToCartAndWishlist context.
- Added cart functionality
- Added wishlist functionality.
chrome-capture-2022-2-28

### CODE REVIEW

File: src/components/cart-item-card/CartItem.jsx [hunk 1]
```diff
@@ -3,38 +3,84 @@ import mensKurta from "../../assets/kurta.png";
...
       </div>
       <div className="cart__item-actions flex-center">
         <div className="item__count">
-          <div className="count__delete bg-secondary p-sm border-rounded-lg">
+          <button
+            className={`count__delete bg-secondary btn border-rounded-lg ${
+              qty <= 1 && "btn-disabled"
+            }`}
+            onClick={(e) => decrementItemQuantity(e, _id)}
+            disabled={qty <= 1}
```
Comment (nishasen) [inline src/components/cart-item-card/CartItem.jsx:63]: I like how you are disabling the button

**Review by nishasen — APPROVED** [review]
I like how you have made the hooks for calling Api, well structured code.

File: src/utils/priceAndQuantityHandler.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  const getDiscountedPrice = (cartItems) => {
+    return cartItems.reduce((acc, curr) => {
+      return (acc = acc + Number(curr.discountedPrice));
+    }, 0);
+  };
+  const getTotalQuantity = (cartItems) => {
+    return cartItems.reduce((acc, curr) => {
+      return (acc = acc + Number(curr.qty));
+    }, 0);
+  };
```
Comment (sadathshariff) [inline src/utils/priceAndQuantityHandler.js:16]: Suggestion to use single reduce to get all the values

**Review by sadathshariff — APPROVED** [review]
Looks Good, like how you have made useAxios Hook to handle all the methods

File: src/utils/priceAndQuantityHandler.js (hunk 2 above)
Comment (rdshinde) [reply src/utils/priceAndQuantityHandler.js:16]: Ya I can do that.

**Pull request merged**